SmartDeviceLink/SDLFileManager.m

```diff
@@ -347,9 +342,9 @@ - (float)sdl_uploadPercentage:(float)totalBytes uploadedBytes:(float)uploadedByt
 }
 
 - (void)uploadFile:(SDLFile *)file completionHandler:(nullable SDLFileManagerUploadCompletionHandler)handler {
-    if (file == nil) {
+    if (file == nil || file.data == nil || file.data.length == 0) {
```

Comment:
You only have to check data length here, if data is nil it will return 0.

Reply:
Fixed